pkg/network/ebpf/c/protocols/http2/decoding.h

```diff
@@ -537,16 +537,14 @@ static __always_inline bool get_first_frame(struct __sk_buff *skb, skb_info_t *s
 // - HEADERS frames
 // - RST_STREAM frames
 // - DATA frames with the END_STREAM flag set
-static __always_inline __u16 find_relevant_frames(struct __sk_buff *skb, skb_info_t *skb_info, http2_frame_with_offset *frames_array, __u8 original_index, http2_telemetry_t *http2_tel) {
+static __always_inline void find_relevant_frames(struct __sk_buff *skb, skb_info_t *skb_info, http2_tail_call_state_t *iteration_value, http2_telemetry_t *http2_tel) {
     bool is_headers_or_rst_frame, is_data_end_of_stream;
     http2_frame_t current_frame = {};
 
-    // We may have found a relevant frame already in http2_handle_first_frame,
-    // so we need to adjust the index accordingly. We do not set
-    // interesting_frame_index to original_index directly, as this will confuse
-    // the verifier, leading it into thinking the index could have an arbitrary
-    // value.
-    __u16 interesting_frame_index = original_index == 1;
+   // If we have found enough interesting frames, we should not process any new frame.
+   if (iteration_value->frames_count >= HTTP2_MAX_FRAMES_ITERATIONS) {
```

Comment:
Let's add a comment explaining how that can be possible, as it can look weird for the naive reader